esp-hal/src/dma/mod.rs

```diff
@@ -2996,17 +2895,20 @@ pub(crate) mod asynch {
             if self.rx.is_done() {
                 self.rx.clear_interrupts();
                 Poll::Ready(Ok(()))
-            } else if self.rx.has_error()
-                || self.rx.has_dscr_empty_error()
-                || self.rx.has_eof_error()
-            {
+            } else if !self.rx.available_in_interrupts().is_disjoint(enum_set!(
+                DmaRxInterrupt::DescriptorError
+                    | DmaRxInterrupt::DescriptorEmpty
+                    | DmaRxInterrupt::ErrorEof
+            )) {
                 self.rx.clear_interrupts();
                 Poll::Ready(Err(DmaError::DescriptorError))
             } else {
-                self.rx.listen_eof();
-                self.rx.listen_in_descriptor_error();
-                self.rx.listen_in_descriptor_error_dscr_empty();
-                self.rx.listen_in_descriptor_error_err_eof();
+                self.rx.listen_in(enum_set!(
```

Comment:
Does this, and a few lines above, need the macro?

Reply:
removed. Seems like the macro is mostly useful in `const` contexts.